mvc_controller.cpp

```diff
@@ -331,19 +331,27 @@ void MvcController::ConditionallyEnableItems
         }
     else if(itembox)
         {
-        // WX !! wxWindowUpdateLocker doesn't seem to help much.
-        wxWindowUpdateLocker u(&control);
-        itembox->Clear();
-        // WX !! Append(wxArrayString const&) "may be much faster"
-        // according to wx online help, but that seems untrue: its
-        // implementation just uses a loop.
+        std::vector<wxString>& itembox_cache = itemboxes_cache_[name];
+
+        std::vector<wxString> items;
         for(int j = 0; j < datum->cardinality(); ++j)
             {
             if(datum->is_allowed(j))
                 {
-                itembox->Append(datum->str(j));
+                items.push_back(datum->str(j));
                 }
             }
+
+        // WX !! wxWindowUpdateLocker doesn't seem to help much.
```

Comment:
Is this comment still correct? I.e. does the time stay the same if you remove this line? If so, we should probably just remove it.

If not, we should move it inside the `if` below as there is no reason to lock the control for updates if we're not going to update it.

Reply:
Ok, I'll measure time for this.